src/Plugin.php

```diff
@@ -100,11 +100,15 @@ function ($code, $message, $file, $line, $context = null) use (&$previousHandler
                         if (PHP_VERSION_ID >= 80000) {
                             $trace = debug_backtrace();
                             foreach ($trace as $idx => $traceEntry) {
-                                if ($traceEntry['function'] !== 'trigger_error') {
+                                if ($traceEntry['function'] !== 'deprecationWarning') {
```

Comment:
Don't we also want to catch `trigger_error` calls? Application or Library code could be using `trigger_error('...', E_USER_DEPRECATED)` and it would be good if debug kit handled that situation.

Reply:
no change in behavior of `trigger_error()` that triggered out of `deprecationWarning()` just like php < 8.0
it means it shows line of calling `trigger_error()`

```php
// src/Controller/AController.php

line 29
line 30 trigger_error("deprecation message", E_USER_DEPRECATED);
line 31

```
//result in both ` php < 8.0` and `> 8.0`
```
App

    APP/Controller/AController.php:30
    deprecation message
```

but before this PR , it was wrong

Reply:
do you say we can add `1` as offset for all `trigger_error()` that triggered out of `deprecationWarning()` ?

Reply:
signature of `deprecationWarning` now (compatible of cake 4) is : 
```
function deprecationWarning(string $message, int $stackFrame = 1)
```
with checking  the `['args'][1]` we want to check used `stackFrame` value 
in direct usage of  `trigger_error` , we dont have this info

Reply:
ok. I was concerned that we would be capturing the incorrect line number for bare `trigger_error()` calls. It sounds like that is not the case though.